api/v1beta1/azuremanagedcontrolplane_default.go

```diff
@@ -96,18 +98,51 @@ func (m *AzureManagedControlPlane) setDefaultSubnet() {
 	}
 }
 
-func setDefaultSku(sku *AKSSku) *AKSSku {
-	if sku == nil {
-		return &AKSSku{
-			Tier: FreeManagedControlPlaneTier,
+func setDefaultSku(sku *AKSSku) {
+	if sku != nil && sku.Tier == PaidManagedControlPlaneTier {
+		sku.Tier = StandardManagedControlPlaneTier
+		ctrl.Log.WithName("AzureManagedControlPlaneWebHookLogger").Info("Paid SKU tier is deprecated and has been replaced by Standard")
+	}
+}
+
+func setDefaultNetworkPlugin(networkPlugin *string) {
+	if networkPlugin == nil {
+		networkPlugin = ptr.To(AzureNetworkPluginName)
+	}
+}
+
+func setDefaultLoadBalancerSKU(loadBalancerSKU *string) {
+	if loadBalancerSKU == nil {
+		loadBalancerSKU = ptr.To("Standard")
```

Comment:
For things this simple, is there a kubebuilder annotation we can add to the type instead of writing code to default the value?